src/app/components/FormComponent/Form.tsx

```diff
@@ -0,0 +1,146 @@
+'use client';
+
+import './Form.scss';
+
+import { useState, useEffect } from 'react';
+import { z } from 'zod';
+
+import { FormData } from '@/app/types/zod';
+import Button from '@/app/components/Button/Button';
+import { Input } from '@/app/components/InputComponent/Input';
+import { Checkbox } from '@/app/components/CheckboxComponent/Checkbox';
+import { FormInitialData } from '@/app/typrs/forminitialData';
+import { initialData } from '@/app/components/FormComponent/helper';
+import { ZodErrorMessage } from '@/app/types/zodErrorMessage';
+
+const formDataSchema: ZodType<FormData> = z.object({
+  fullname: z.string().min(5, 'Error name').trim(),
+  email: z.string().email('Error email format').trim(),
+});
+
+interface Props {
+  onClose: () => void;
+}
+
+export const Form: React.FC<Props> = ({ onClose }) => {
+  const [formData, setFormData] = useState<FormInitialData>(initialData);
+  const [errors, setErrors] = useState<ZodErrorMessage | []>([]);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [btnColor, setBtnColor] = useState('primary');
```

Comment:
trochę źle zrobiłam, że dodałam wtedy to, co prosiłeś z buttonami a 100% na desktopie. Dla 100% na desktopie variantem powinien być secondary, a primary to jest taka, co np mamy na home. Już to zmieniłam w buttonach więc zmień też na secondary